### DISCUSSION TITLE
why does cumulative_hospitalized increase from 0 to 1 before severe increases from 0 to 1

### DISCUSSION THREAD

**Discussion by kaiserso** [neherlab/covid19_scenarios#398] [General] [answered]
Hi,

Great tool - thank you for providing.

I believe cumulative_hospitalized is the cumulative number of severe cases. However, in all of my scenarios I noted that cumulative_hospitalized increased from zero to one while severe remained at zero for about 10 days, then increased to two when severe increased to one, etc.  Is this due to a type of rounding issue?

Second question - I assume the categories are mutually exclusive (severe, exposed, overflow, ICU, infectious, susceptible) but the cumulative categories are not necessarily mutually exclusive because individuals can move from hospitalized to critical and hence get counted in both cumulative counts?

thanks!
Peter

**Comment by noleti** [comment]
I have the feeling we might be accumulating the data twice, but I am unfamiliar with the code. In particular, https://github.com/neherlab/covid19_scenarios/blob/21d9672eca7acab645d7bf6535281199f0cbee6b/src/algorithms/model.ts#L473 looks like we are aggregating data that is already accumulative. I would expect that we can directly copy `d.cumulative[k]` into `tp.cumulative[k].total`

@nnoll I think you wrote that code, can you comment? We likely want to get rid of total here completely, and write the .cumulative to file directly in 
https://github.com/neherlab/covid19_scenarios/blob/2498f1e357590beb159eb8ed8a49a324e9ed1f3b/src/algorithms/model.ts#L531

/edit: after more reading of the code, it seems that the reduce() is over the different age keys in d.cumulative (summing them up to a total). So that seems to be correct after all, my bad!

Regarding the second question: I share your understanding that cumulative is not mutually exclusively counting individual people.

**Comment by noleti** [comment]
Sorry for the confusion, I dove deeper into the code. I believe that this might be a case where the single severe=hospitalized patient recovered within the same day. In particular, the count of severe per day is the following: 

```
      flux.infectious.severe[age] +
        flux.critical.severe[age] +
        flux.overflow.severe[age] -
        flux.severe.critical[age] -
        flux.severe.recovered[age]
```

whereas the `cumulative_hospitalized` is just oldval+`flux.infectious.severe[age]`. So if the  flux.severe.recovered[age]=1, and flux.infectious.severe[age]=1 (lets assume others are 0), then severe would be 0 and cumulative_hospitalized would be 1.

**Comment by nnoll** [comment, marked as answer]
I think (without being able to simulate your exact parameters for a deeper dive) @noleti's diagnosis/hypothesis is correct. As we report statistics per day and the residence time in the hospital is (for now at least) assumed to be exponentially distributed, there will be many cases that enter the hospital and then immediately leave that day (they are given 20 chances to leave per day as written now). Thus hospitalization will register the admission but that day's statistics won't as they were "immediately" discharged from the standpoint of exported data. This is apparent when numbers are low.

To answer your second question, all "current" statistics, i.e. numbers that report on the current number of particular compartments, are indeed mutually exclusive. Cumulative numbers don't have this property for a few reasons:  (i) all patients in critical care were first admitted to the hospital, (ii) all fatalities were first in the ICU/overflow, (iii) due to the way transitions occur, an individual can progress back to critical care once they previously "recovered" (although this should occur very infrequently).  Taking the difference of these categories can give you a "delta" which should tell you the cumulative number of individuals that had the conjugate outcome - i.e. hospitalizations - critical tell you the discharged individuals.

Hope this helps!

**Reply by kaiserso** [reply]
Thank you @noleti for the investigation and @nnoll for the confirmation.  Very helpful and makes a lot of sense.
